python/main.py

```diff
@@ -18,15 +21,60 @@
     allow_headers=["*"],
 )
 
+"""
+Gets the list of all items
+"""
 @app.get("/")
 def root():
-    return {"message": "Hello, world!"}
+    # get the list of all items in the database
+    items = database.get_items()
+    # format the list and return
+    return format_items(items)
 
+"""
+Gets item with the given item_id
+"""
+@app.get("/items/{item_id}")
+async def get_item_by_id(item_id):
+
+    item = database.get_id_by_id(item_id)
+    if len(item) == 0:
+        logger.debug(f"Item not found: {item_id}")
+        return f"Item not found: {item_id}"
+    
+    return format_items(item)
+
+
+"""
+Create a new item with the given name, cateogry, image
+"""
 @app.post("/items")
-def add_item(name: str = Form(...)):
-    logger.info(f"Receive item: {name}")
+def add_item(name: str = Form(...), category: str = Form(...), image: str = Form(...)):
```

Comment:
In the instruction, the request looks like this with "file" option (-f):
```
curl -X POST \
  --url 'http://localhost:9000/items' \
  -F 'name=jacket' \
  -F 'category=fashion' \
  -F 'image=@images/local_image.jpg'
```
But at the moment my program accepts only requests like this with "data" option (-d):

```
curl -X POST \
  --url 'http://localhost:9000/items' \
  -d 'name=jacket' \
  -d 'category=fashion' \
  -d 'image=@images/default.jpg'
```
I wonder what's the difference between -d and -F. (googled it, but didn't get well)
and which part of code I should change to accept in -F form.

Reply:
<img width="1031" alt="Screen Shot 2022-05-17 at 15 16 00" src="https://user-images.githubusercontent.com/12114213/168782467-29346b12-7755-45e6-987f-af782f8de163.png">

Please check this manual.

Reply:
Thank you! I understood that -d accepts string and -F accepts files!